ntoskrnl/ob/obname.c

```diff
@@ -695,7 +679,12 @@ ObpLookupObjectName(IN HANDLE RootHandle OPTIONAL,
         else
         {
 ParseFromRoot:
-            /* FIXME: Check if we have a device map */
+            /* Deference the device map if we already have one */
+            if (DeviceMap != NULL)
+            {
+                ObfDereferenceDeviceMap(DeviceMap);
+                DeviceMap = NULL;
+            }
 
             /* Check if this is a possible DOS name */
```

Comment:
Just do `LocalName = *ObjectName` here (or even at the start of the function) to avoid having to have it in all the branches?